catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreAutoPublisher.java

```diff
@@ -0,0 +1,146 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.api.impl;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.codice.ddf.registry.api.RegistryStore;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.federationadmin.service.RegistryPublicationService;
+import org.codice.ddf.security.common.Security;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventConstants;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.source.SourceMonitor;
+
+public class RegistryStoreAutoPublisher implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryStoreAutoPublisher.class);
+
+    private FederationAdminService federationAdminService;
+
+    private RegistryPublicationService registryPublicationService;
+
+    private ConcurrentHashMap<String, Boolean> registryStoreMap = new ConcurrentHashMap<>();
+
+    public void setFederationAdminService(FederationAdminService federationAdminService) {
+        this.federationAdminService = federationAdminService;
+    }
+
+    public void setRegistryPublicationService(
+            RegistryPublicationService registryPublicationService) {
+        this.registryPublicationService = registryPublicationService;
+    }
+
+    public void bindRegistryStore(ServiceReference reference) {
+
+        BundleContext bundleContext = getBundleContext();
+
+        if (reference != null && bundleContext != null) {
+            RegistryStore registryStore = (RegistryStore) bundleContext.getService(reference);
+
+            Boolean previousState = registryStoreMap.put(registryStore.getConfigurationPid(), registryStore.isAutoPush());
+
+            SourceMonitor storeRegisteredSourceMonitor = new SourceMonitor() {
+                @Override
+                public void setAvailable() {
+                    if (registryStore.isAutoPush()) {
+                        registryPublish(registryStore);
+                    }
+                }
+
+                @Override
+                public void setUnavailable() {
+                }
+            };
+
+            if (registryStore.isAvailable(storeRegisteredSourceMonitor) && registryStore.isAutoPush()) {
+                if(previousState == null || !previousState) {
+                    registryPublish(registryStore);
+                }
+            }
+        }
+    }
+
+    void registryPublish(RegistryStore registryStore) {
+
+        if (!registryStore.getRegistryId()
```

Comment:
Invert if condition and return; That will reduce cyclomatic complexity of this method.